azurerm/internal/services/digitaltwins/digital_twins_endpoint_eventhub_resource.go

```diff
@@ -0,0 +1,176 @@
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing Digital Twins Endpoint %q (Resource Group %q / Instance %q): %+v", name, digitalTwinsId.ResourceGroup, digitalTwinsId.Name, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_digital_twins_endpoint_eventhub", id)
+		}
+	}
+
+	properties := digitaltwins.EndpointResource{
+		Properties: &digitaltwins.EventHub{
+			EndpointType:                 digitaltwins.EndpointTypeEventHub,
+			ConnectionStringPrimaryKey:   utils.String(d.Get("eventhub_primary_connection_string").(string)),
+			ConnectionStringSecondaryKey: utils.String(d.Get("eventhub_secondary_connection_string").(string)),
+			DeadLetterSecret:             utils.String(d.Get("dead_letter_storage_secret").(string)),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, digitalTwinsId.ResourceGroup, digitalTwinsId.Name, name, properties)
+	if err != nil {
+		return fmt.Errorf("creating/updating Digital Twins Endpoint EventHub %q (Resource Group %q / Instance %q): %+v", name, digitalTwinsId.ResourceGroup, digitalTwinsId.Name, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of the Digital Twins Endpoint EventHub %q (Resource Group %q / Instance %q): %+v", name, digitalTwinsId.ResourceGroup, digitalTwinsId.Name, err)
+	}
+
+	if _, err := client.Get(ctx, digitalTwinsId.ResourceGroup, digitalTwinsId.Name, name); err != nil {
+		return fmt.Errorf("retrieving Digital Twins Endpoint EventHub %q (Resource Group %q / Instance %q): %+v", name, digitalTwinsId.ResourceGroup, digitalTwinsId.Name, err)
+	}
+
+	d.SetId(id)
+
+	return resourceDigitalTwinsEndpointEventHubRead(d, meta)
+}
+
+func resourceDigitalTwinsEndpointEventHubRead(d *schema.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).DigitalTwins.EndpointClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DigitalTwinsEndpointID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.DigitalTwinsInstanceName, id.EndpointName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Digital Twins Event Hub Endpoint %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Digital Twins Endpoint EventHub %q (Resource Group %q / Instance %q): %+v", id.EndpointName, id.ResourceGroup, id.DigitalTwinsInstanceName, err)
+	}
+	d.Set("name", id.EndpointName)
+	d.Set("digital_twins_id", parse.NewDigitalTwinsInstanceID(subscriptionId, id.ResourceGroup, id.DigitalTwinsInstanceName).ID(""))
+	if resp.Properties != nil {
```

Comment:
Should we pull the values for primary connection string, secondary connection string, and dead letter secret from the passed schema.ResourceData to store them in the state file here? I am guessing the API doesn't return those values do they?

Reply:
No, the api doesn't return. I think it's not necessary to refill the values in the first level of schema.(In blocks, we'll need the refill the value got from schema). If we don't set the value from get response, it'll keep its initial value set by the user. But we need to skip these values in "terraform import" test. I did so previously, am I right?

Reply:
Yep, you are correct, top level values are automatically written to the state file so pulling them from the schema.ResourceData should not be necessary in this case. 🙂